producer.go

```diff
@@ -112,10 +117,27 @@ func (p *Producer) Produce(message []byte, opts ...ProduceOpt) error {
 
 }
 
+func (opts *ProduceOpts) validateHeaderKey() error {
+	for i, _ := range opts.Header {
+		if strings.HasPrefix(i, "$memphis") {
+			return errors.New("Keys in headers should not start with $memphis")
+		}
+	}
+	return nil
+}
+
 // ProducerOpts.produce - produces a message into a station using a configuration struct.
 func (opts *ProduceOpts) produce(p *Producer) error {
+	memphisHeader := map[string][]string{"$memphisconnectionId": {p.conn.ConnId}, "$memphisproducedBy": {p.Name}}
+
+	err := opts.validateHeaderKey()
+	if err != nil {
+		return err
+	}
+	maps.Copy(memphisHeader, opts.Header)
+
 	natsMessage := nats.Msg{
-		Header:  map[string][]string{"connectionId": {p.conn.ConnId}, "producedBy": {p.Name}},
+		Header:  memphisHeader,
```

Comment:
not related to here but when you add something new you are the responsible to add it into the readme file as well otherwise users won't be able to use it